docs/rfc/rfc-1.adoc

```diff
@@ -180,12 +179,48 @@ simple as "these funds can be spent by wallet 0xabc", or in our case, as
 complex as "these funds can be spent by wallet 0xabc but if they aren't spent
 within 30 days they can be spent by wallet 0x123". This gives us the ability to
 create deposits that automatically are refunded after 30 days if they aren't
-swept. Thus, if a user misfunds or they get cold feet (for any reason), all
-they need to do is not submit their reveal and wait 30 days.
+<<sweeping,swept>>. Thus, if a user misfunds or they get cold feet (for any
+reason), all they need to do is not submit their reveal and wait 30 days.
+
+[[sweeping]]
+==== Sweeping
+The operators sign a transaction that unlocks all revealed deposits, combines
+them into a single UTXO with the existing UTXO, and relocks that transactions
+without a 30-day refund clause to same wallet. This has two main effects: it
+consolidates the UTXO set and it disables the refund.
+
+This process is called a "sweep", and occurs based on a governable frequency.
+Any deposit below a governable threshold is ignored.
+
+After submitting the transaction, it will cost some amount of bitcoin based on
+what miners are charging for the bitcoin fee. This fee comes out of the unswept
+deposits. When the transaction clears, the users unswept balances are
+decreased, and their swept balances are increased (after paying their share of
+the mining fee).
+
+The main downside to this approach is that it can take, in the worst case, up
+to the frequency parameter in time for a user to be able to mint TBTC. To help
+alleviate this, two suggestions:
+
+1) We surface when the next scheduled sweep is somewhere in the dApp. This
+allows users to feel a lot better about when sweeps are happening, and feel
+better about when their funds will be available. There is also something to be
+said about the marketing around explaining that we're batching in order to
+reduce fees across the board for the end user, which allows for the decentralized
+product to compete with the centralized ones.
+
+2) We allow users to request that their TBTC is minted as soon as they have a
+swept account balance. This makes it so they don't have to wait, check, and
+come back later and mint.
+
+Combining these ideas, a user would deposit some BTC, reveal their eth address
+and blinding factor, and then request that TBTC gets minted ASAP. Checking the
+dApp, they can see that they should expect TBTC in their provided wallet
+address in 3 hours with no further interaction.
 
```

Comment:
There is one more thing that came up in earlier discussions and is worth thinking about: `CHECKLOCKTIMEVERIFY` requirements. One may reveal their ETH address and blinding factor just a moment before BTC script unlocks the funds for refund. We should have a requirement regarding the time left for the wallet to execute the sweep. If that requirement is not met deposit is ignored.

Now, there is a question where and how this requirement is checked. Is it possible to check it during the reveal transaction? (I doubt it). Should the wallet check it when sweeping and not even try to sweep such a deposit?

Reply:
> There is one more thing that came up in earlier discussions and is worth thinking about: CHECKLOCKTIMEVERIFY requirements. One may reveal their ETH address and blinding factor just a moment before BTC script unlocks the funds for refund. We should have a requirement regarding the time left for the wallet to execute the sweep. If that requirement is not met deposit is ignored.

Maybe I'm missing something obvious. Say that they do this, and they reveal. Now, they *could* refund their BTC, but we can also still sweep. If they refund their BTC, we can't sweep. They'd have to mine the transaction in between when we start the signing process and when we go to submit the transaction, and for what gain? (I don't actually know what happens if we submit a big batch transaction and one of the inputs is already spent)

If they don't refund, then we can proceed as normal!

You're totally right in that if we put in a time limit like "we'll only process stuff that has >1 day left on the cltv" then we don't run into this nonsense, so it might be worth just nipping in the bud